addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/ExtensionReplacerTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.replacer;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.REQ_BODY_STR;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.REQ_HEADER;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.REQ_HEADER_STR;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.RESP_BODY_STR;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.RESP_HEADER;
+import static org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType.RESP_HEADER_STR;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class ExtensionReplacerTest {
+
+    private static final String MATCHING_STRING_WITH_HEX_BYTE =
+            new String(new byte[] {'a', 'b', 'c', 1, 3, 2, 'd', 'e', 'f'});
+    private static final String REPLACED_STRING_WITH_BINARY_VALUE =
+            new String(new byte[] {'a', 'b', 'c', 1, 2, 3, 'd', 'e', 'f'});
+    private HttpMessage msg;
+
+    @Before
+    public void setUp() {
+        msg = new HttpMessage();
+    }
+
+    @Test
+    public void shouldReplaceHeaderByHexValueInRequest() throws HttpMalformedHeaderException {
+        // Given
+        ExtensionReplacer extensionReplacer = given_a_hex_byte_replacement_rule_for(REQ_HEADER);
+
+        msg.setRequestHeader("GET / HTTP/1.1\r\n" + "X-CUSTOM: " + MATCHING_STRING_WITH_HEX_BYTE);
```

Comment:
The `X-CUSTOM` header could be removed, it doesn't affect the outcome of the test.